From 0054720e536b3bca12d0a9573554a180305d09c5 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 23 May 2019 14:31:51 -0700
Subject: [PATCH] TRITON-1699 sdc-amon workaround to build on x86_64 broke
 'make check' on non-pkgsrc-using envs Reviewed by: Chris Burroughs
 <chris.burroughs@joyent.com> Approved by: Chris Burroughs
 <chris.burroughs@joyent.com>

---
 Makefile | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 64655ca..9f74d63 100644
--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,16 @@ CLEAN_FILES += agent/node_modules relay/node_modules \
 	./node_modules test/node_modules build/amon-*.tgz \
 	build/amon-*.tar.gz lib build/pkg
 
+# These files are transformed during setup/configure. We're
+# listing them here just so that check-manifests can be run
+# on the pre-converted versions, which are still valid xml.
+SMF_MANIFESTS	= \
+	agent/smf/manifests/amon-agent.xml.in \
+	relay/smf/manifests/amon-relay.xml.in \
+	relay/smf/manifests/amon-zoneevents.xml.in \
+	master/smf/amon-master.smf.in
+
+
 # The prebuilt sdcnode version we want. See
 # "tools/mk/Makefile.node_prebuilt.targ" for details.
 NODE_PREBUILT_VERSION=v0.8.28
@@ -67,10 +77,16 @@ TOP ?= $(error Unable to access eng.git submodule Makefiles.)
 # node-trace-provider fix.  This hack should be removed if amon is every updated
 # past nodejs 0.8.  See also TRITON-1658
 #
-# NOTE: This PATH definition must appear before node_prebuilt, which also mucks
-# with PATH.
+# Notes:
+# - We cannot add a "PATH=..." change to NPM_ENV because
+#   Makefile.node_prebuilt.* is already doing so.
+# - Only make this PATH addition on SunOS where we know we have
+#   /opt/local/bin/gcc, otherwise other parts of the build using gcc on
+#   non-SunOS (e.g. 'make check' builds of jsl) break.
 #
-export PATH:=$(TOP)/tools/m32-gcc-wrapper:$(PATH)
+ifeq ($(shell uname -s),SunOS)
+	export PATH:=$(TOP)/tools/m32-gcc-wrapper:$(PATH)
+endif
 
 ifeq ($(shell uname -s),SunOS)
        include deps/eng/tools/mk/Makefile.node_prebuilt.defs
-- 
2.21.0

